Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/ScheduledRides/ScheduledRideAssignedOnUpdate.hs

```diff
@@ -264,3 +292,23 @@ cancelOrReallocate ride cReason isForceReallocation req = do
           }
   (_cancellationCnt, _isGoToDisabled) <- RideCancel.cancelRideImpl RideCancel.cancelRideHandle req ride.id cancelReq isForceReallocation
   pure ()
+
+data Result a b = APIFailed | DistanceResp (GetDistanceResp a b)
+
+errorCatchAndHandle ::
+  ( ServiceFlow m r,
+    HasCoordinates a,
+    HasCoordinates b
+  ) =>
+  [TMaps.GetDistanceReq a b] ->
+  ( TMaps.GetDistanceReq a b ->
+    m (GetDistanceResp a b)
+  ) ->
+  m [Result a b]
+errorCatchAndHandle reqs func = foldM processRequest [] reqs
```

Comment:
`foldM processRequest [] reqs
  where
    processRequest (stopExec, acc) req = do
     if stopExec then return (stopExec, acc) else do
      resp <- try @_ @SomeException $ func req
      case resp of
        Left _ -> return (True, [APIFailed])
        Right result -> return (False, DistanceResp result : acc)`